src/esp/assets/MeshMetaData.h

```diff
@@ -5,10 +5,48 @@
 #pragma once
 
 #include "esp/core/esp.h"
+#include "esp/gfx/magnum.h"
 
 namespace esp {
 namespace assets {
 
+//! Some mesh files include a transformation hierarchy. A @ref MeshTransformNode
+//! stores this hierarchy and indices for the meshes at each level such that it
+//! can be reused to instances meshes later.
+struct MeshTransformNode {
+  //! mesh ID within @ref MeshMetaData::meshIndex
+  int meshIDLocal;
+
+  //! material ID within @ref MeshMetaData::materialIndex
+  int materialIDLocal;
+
+  //! Object ID in the original file
+  int componentID;
+
+  std::vector<MeshTransformNode> children;
+
+  //! Node local transform to the parent frame
+  Magnum::Matrix4 T_parent_local;
+
+  MeshTransformNode() {
+    meshIDLocal = ID_UNDEFINED;
+    materialIDLocal = ID_UNDEFINED;
+    componentID = ID_UNDEFINED;
+  };
+
+  //! copy constructor which duplicates the @ref MeshTransformNode tree of which
+  //! val is the root.
+  MeshTransformNode(const MeshTransformNode& val) {
+    componentID = val.componentID;
+    meshIDLocal = val.meshIDLocal;
+    materialIDLocal = val.materialIDLocal;
+    T_parent_local = Magnum::Matrix4(val.T_parent_local);
```

Comment:
What's the reason to use `snake_case` with first letter upper-case? To me that's quite confusing when reading the code.

Reply:
It's not really `snake_case`, it is to denote subscripting like in latex.   I proposed this naming scheme for transforms as I really like it :) It is `T_<to>_<from>` as you can then write something like `pt_<to> = T_<to>_<from> pt_<from>` and you can think of it as the `<from>`s "canceling" out.  Same thing with `T_<a>_<c> = T_<a>_<b> * T_<b>_<c>`.  But it does break normal code style.

Reply:
Yep, `transformFromLocalToParent` is also fine by me :) -- I use `T_parent_local` and have grown to really like it (although I do admit remembering the notation took a couple tries), but it is merely a suggestion and is localized enough that we can change it!